packages/cache-service/src/data-packages/data-packages.service.ts

```
@@ -36,6 +37,48 @@ export interface StatsRequestParams {
 
 @Injectable()
 export class DataPackagesService {
+  private readonly logger = new Logger(DataPackagesService.name);
+
+  constructor(private readonly bundlrService: BundlrService) {}
+
+  /**  Save dataPackages to DB and bundlr if enabled */
+  async saveMany(
+    dataPackagesToSave: CachedDataPackage[],
+    nodeEvmAddress: string
+  ): Promise<void> {
+    const savePromises: Promise<any>[] = [];
+    const saveToDbPromise = this.saveManyDataPackagesInDB(dataPackagesToSave)
+      .catch(() =>
```

Quite verbose code, however I didn't see added value in extracting it to new function to many questions started to come up

---

However I'm open to changes

---

What do you think about creating a helpful function to wrap promises with log messages, e.g. smth like this
```js
// pseudo-code
const addLogs = (promise: Promise<void>, promiseName: string): Promise<void> => {
   return promise.then(() => console.log(`${promiseName} completed}).catch(err => console.err(`${promiseName} failed. ${stringifyErr(err)}`);
}
```

---

I agree good idea, however, I think that addLogs function should rethrow the error after logging it, and handling of the error should happen where business logic happens. Also, the name doesn't signalize that function will consume error.